File: TAM.LogisticSystem/client/js/Services/MaintenanceKonfigurasiExportFileDccpService.ts

export class MaintenanceKonfigurasiExportFileDccpService {
    static $inject = ["$http"];
    constructor(httpClient: angular.IHttpService) {
        this.httpClient = httpClient;
    }
    httpClient: angular.IHttpService;
    //getAllData Service
    getAllData() {
        return this.httpClient.get<MaintenanceKonfigurasiExportFileDccpModel[]>("api/v1/MaintenanceKonfigurasiExportFileDccpAPI");
    }

    //delete selected 
    deleteSelectedData(id: number) {
        return this.httpClient.delete("api/v1/MaintenanceKonfigurasiExportFileDccpAPI/" + id);
    }

    //post new config service
    postData(postModel: MaintenanceKonfigurasiExportFileDccpPostModel) {
        return this.httpClient.post("api/v1/MaintenanceKonfigurasiExportFileDccpAPI", postModel);
    }

    //update config service
    updateData(updateModel: MaintenanceKonfigurasiExportFileDccpModel) {
        console.log(updateModel);
        return this.httpClient.put("api/v1/MaintenanceKonfigurasiExportFileDccpAPI", updateModel);
    }
}
export class MaintenanceKonfigurasiExportFileDccpModel {
    id: number;
    description: string;
    rangeStart: string;
    rangeEnd: string;
}
export class MaintenanceKonfigurasiExportFileDccpPostModel {
    description: string;
    rangeStart: string;
    rangeEnd: string;
}

File: takeout/December/Services/MaintenanceKonfigurasiExportFileDccpService.cs
using Dapper;
using Microsoft.EntityFrameworkCore;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using TAM.LogisticSystem.Entities;
using TAM.LogisticSystem.Models;

namespace TAM.LogisticSystem.Services
{
    public class MaintenanceKonfigurasiExportFileDccpService
    {
        private readonly LogisticDbContext context;

        public MaintenanceKonfigurasiExportFileDccpService(LogisticDbContext dbContext)
        {
            this.context = dbContext;
        }
        /// <summary>
        /// get all config
        /// </summary>
        /// <returns></returns>
        public async Task<List<MaintenanceKonfigurasiExportFileDccpViewModel>> GetAllConfig()
        {
            var allConfig = (await this.context.Database.GetDbConnection().QueryAsync<MaintenanceKonfigurasiExportFileDccpViewModel>($@"Select 
            DailyCarCarrierPlanExcelConfigurationId  as Id,
            Description,
            RangeStart,
            RangeEnd 

            From DailyCarCarrierPlanExcelConfiguration")).ToList();
            return allConfig;
        }


        /// <summary>
        /// add new config
        /// </summary>
        /// <param name="addedMOdel"></param>
        /// <returns></returns>
        public async Task AddConfig(MaintenanceKonfigurasiExportFileDccpPostViewModel addedMOdel)
        {
            var newAddedEntities = new DailyCarCarrierPlanExcelConfiguration();
            newAddedEntities.Description = addedMOdel.Description;
            newAddedEntities.RangeEnd = addedMOdel.RangeEnd;
            newAddedEntities.RangeStart = addedMOdel.RangeStart;
            this.context.DailyCarCarrierPlanExcelConfiguration.Add(newAddedEntities);
            await this.context.SaveChangesAsync();
        }

        /// <summary>
        /// Delete selected Config
        /// </summary>
        /// <param name="id"></param>
        /// <returns></returns>
        public async Task<int> DeleteSingleData(int id)
        {   
            var selectedData = await this.context.DailyCarCarrierPlanExcelConfiguration.Where(Q => Q.DailyCarCarrierPlanExcelConfigurationId == id).FirstOrDefaultAsync();
            this.context.Remove(selectedData);
            return await this.context.SaveChangesAsync();
        }


        /// <summary>
        /// Update Selected Data Service
        /// </summary>
        /// <param name="updateMOdel"></param>
        /// <returns></returns>
        public async Task UpdateData(MaintenanceKonfigurasiExportFileDccpViewModel updateMOdel)
        {
            var selectedUpdateModel = await this.context.DailyCarCarrierPlanExcelConfiguration.Where(Q => Q.DailyCarCarrierPlanExcelConfigurationId == updateMOdel.Id).FirstOrDefaultAsync();
            selectedUpdateModel.Description = updateMOdel.Description;
            selectedUpdateModel.RangeEnd = updateMOdel.RangeEnd;
            selectedUpdateModel.RangeStart = updateMOdel.RangeStart;

            this.context.DailyCarCarrierPlanExcelConfiguration.Update(selectedUpdateModel);
            await this.context.SaveChangesAsync();
        }
    }
}
